### ISSUE TITLE
Stream inside transaction issue

### ISSUE THREAD

**Issue by gchauvet** [tgriesser/knex#625]
Hi,

I encountered an issue with Knex when I try to execute a SQL query in streaming mode : 
```javascript
it('with direct stream', function (done) {
            builder.transaction(function (tx) { /* builder is a Knex instance */
                async.eachSeries([
                    'SET join_collapse_limit to 1',
                    'SET enable_nestloop = off'
                ],
                function (request, cb) {
                    builder
                    .raw(request)
                    .transacting(tx)
                    .exec(cb);
                },
                function (err) {
                    var stream =
                    builder('products')
                    .transacting(tx)
                    .stream();
            
                    stream.on('end', function () {
                        tx.commit();
                        done();
                    });
                    stream.pipe(JSONStream.stringify());
                });
            })
            .catch(function (err) {
                console.log(err);
            });
        });
   });
```

Returns me this result : 
```
with direct stream :
{ __cid: '__cid1', sql: 'begin;' }
{ __cid: '__cid1',
  sql: 'SET join_collapse_limit to 1',
  method: 'raw',
  bindings: [] }
{ __cid: '__cid1',
  sql: 'SET enable_nestloop = off',
  method: 'raw',
  bindings: [] }
{ __cid: '__cid2',     <---- not same connection, required '__cid1'
  method: 'select',
  options: undefined,
  bindings: [],
  sql: 'select * from "products"' }
{ __cid: '__cid1', sql: 'commit;' }
```

If I execute this query in callback mode :
```
with direct stream :
{ __cid: '__cid1', sql: 'begin;' }
{ __cid: '__cid1',
  sql: 'SET join_collapse_limit to 1',
  method: 'raw',
  bindings: [] }
{ __cid: '__cid1',
  sql: 'SET enable_nestloop = off',
  method: 'raw',
  bindings: [] }
{ __cid: '__cid1',
  method: 'select',
  options: undefined,
  bindings: [],
  sql: 'select * from "products"' }
{ __cid: '__cid1', sql: 'commit;' }
```

**Comment by tgriesser** [comment]
That's no good... but what happens if you swap 

```js
builder('products')
                    .transacting(tx)
                    .stream();
```

with 

```js
tx('products').stream();
```

Also, what version are you using?

**Comment by gchauvet** [comment]
I get the same result with tx('products').stream().
I use lastest stable version : 0.7.3

**Comment by gchauvet** [comment]
Any clue on this issue ? I'm stuck :/

**Comment by gchauvet** [comment]
Okay, below my workaround for this issue :

```javascript
it('with direct stream', function (done) {
            builder.transaction(function (tx) { /* builder is a Knex instance */
                async.eachSeries([
                    'SET join_collapse_limit to 1',
                    'SET enable_nestloop = off'
                ],
                function (request, cb) {
                    builder
                    .raw(request)
                    .transacting(tx)
                    .exec(cb);
                },
                function (err) {
                    var stream =
                    builder('products')
                    .transacting(tx)
                    .connection(tx._runner.connection) // <--- the workaround
                    .stream();

                    stream.on('end', function () {
                        tx.commit();
                        done();
                    });
                    stream.pipe(JSONStream.stringify());
                });
            })
            .catch(function (err) {
                console.log(err);
            });
        });
   });
```

**Comment by tgriesser** [comment]
This should be fixed and tested in 0.8.
